File: packages/typia/src/transformers/features/http/HttpAssertHeadersTransformer.ts
import { HttpAssertHeadersProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpAssertHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.assertHeaders",
      write: HttpAssertHeadersProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/HttpAssertHeadersTransformer.go
package http

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type httpAssertHeadersTransformerNamespace struct{}

var HttpAssertHeadersTransformer = httpAssertHeadersTransformerNamespace{}

func (httpAssertHeadersTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "http.assertHeaders",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativehttpprogrammers.HttpAssertHeadersProgrammer.Write(x)
		},
	})
}
